src/backend/distributed/utils/reference_table_utils.c

```diff
@@ -233,11 +254,11 @@ EnsureReferenceTablesExistOnAllNodesExtended(char transferMode)
 		CloseConnection(connection);
 	}
 
-	/*
-	 * Unblock other backends, they will probably observe that there are no
-	 * more worker nodes without placements, unless nodes were added concurrently
-	 */
-	UnlockColocationId(colocationId, ExclusiveLock);
+	/* release all the locks we acquired for the above operations */
```

Comment:
This comment needs a better explaination.

We should release the lock here because the reference tables have been copied via a loopback connection and updated the metadata.

We do not need to hold the locks anymore. Any other backend running the above checks will find that the reference tables have already been replicated.